metrics-api/src/main/java/software/amazon/swage/metrics/MetricContext.java

```diff
@@ -73,6 +72,25 @@ default void count(Metric label) {
         count(label, 1L);
     }
 
+    /**
+     * Create a child context with the supplied attributes.
+     * The child context will inherited the attributes from this instance and
+     * use the supplied attributes as overrides.
+     *
+     * @param attributes the attribute overrides to apply.
+     * @return a new child context with combined attributes.
+     */
+    MetricContext newChildContext(TypedMap attributes);
```

Comment:
'new' here is not really necessary, and 'context' is somewhat redundant - why not just call this 'child' or similar?